From 8257e52557c488a76ddc73ab416342f367499b58 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Sat, 16 Jul 2016 08:01:16 +0900
Subject: [PATCH 1749/2848] HADOOP-13298. Fix the leftover L&N files in
 hadoop-build-tools/src/main/resources/META-INF/.
 (ozawa)

(cherry picked from commit ea9f43781e5a9a4a6121dd3be106ced6b811754d)
(cherry picked from commit 65adabd0d1bba1c59653cfcd9b4b6253815adb15)
(cherry picked from commit 2f546755c8186c8cc6368074e7290b5418ddb0e2)

Change-Id: Ic3eec9438e67648f2807957ed64ba9f245bb3eaf
---
 hadoop-build-tools/pom.xml |   42 ++++++++++++++++++++++++++++++++++++++++++
 pom.xml                    |   26 --------------------------
 2 files changed, 42 insertions(+), 26 deletions(-)

diff --git a/hadoop-build-tools/pom.xml b/hadoop-build-tools/pom.xml
index b875458..b2d00a4 100644
--- a/hadoop-build-tools/pom.xml
+++ b/hadoop-build-tools/pom.xml
@@ -29,6 +29,19 @@
     <failIfNoTests>false</failIfNoTests>
   </properties>
   <build>
+    <resources>
+      <resource>
+        <directory>${project.basedir}/target/extra-resources</directory>
+        <targetPath>META-INF</targetPath>
+        <includes>
+          <include>LICENSE.txt</include>
+          <include>NOTICE.txt</include>
+        </includes>
+      </resource>
+      <resource>
+        <directory>${project.basedir}/src/main/resources</directory>
+      </resource>
+    </resources>
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
@@ -37,17 +50,46 @@
           <skip>true</skip>
         </configuration>
       </plugin>
+      <!-- copy L&N files to target/extra-resources -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>copy-resources</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.basedir}/target/extra-resources</outputDirectory>
+              <resources>
+                <resource>
+                  <directory>../</directory>
+                  <includes>
+                    <include>LICENSE.txt</include>
+                    <include>NOTICE.txt</include>
+                  </includes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <!-- add entries for L&N files to remote-resources.xml in jar file -->
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-remote-resources-plugin</artifactId>
         <executions>
           <execution>
+            <phase>process-resources</phase>
             <goals>
               <goal>bundle</goal>
             </goals>
           </execution>
         </executions>
         <configuration>
+          <resourcesDirectory>${project.build.outputDirectory}</resourcesDirectory>
           <includes>
             <include>META-INF/LICENSE.txt</include>
             <include>META-INF/NOTICE.txt</include>
diff --git a/pom.xml b/pom.xml
index 005e931..4573acf 100644
--- a/pom.xml
+++ b/pom.xml
@@ -271,32 +271,6 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
         <inherited>true</inherited>
         <extensions>true</extensions>
       </plugin>
-      <plugin>
-        <!-- Copy license and notice files into hadoop-build-tools'
-             metadata, which will then be bundled into jars.
-        -->
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-antrun-plugin</artifactId>
-        <executions>
-          <execution>
-            <id>copy-files</id>
-            <phase>process-resources</phase>
-            <goals>
-              <goal>run</goal>
-            </goals>
-            <configuration>
-              <target>
-                <copy todir="${project.build.directory}/../hadoop-build-tools/src/main/resources/META-INF/">
-                  <fileset dir="${project.build.directory}/../">
-                    <include name="LICENSE.txt"/>
-                    <include name="NOTICE.txt"/>
-                  </fileset>
-                </copy>
-              </target>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
     </plugins>
   </build>
   
-- 
1.7.9.5

